codegen/src/main/java/software/amazon/awssdk/codegen/docs/PaginationDocs.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.codegen.docs;
+
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.CodeBlock;
+import software.amazon.awssdk.codegen.model.intermediate.IntermediateModel;
+import software.amazon.awssdk.codegen.model.intermediate.OperationModel;
+import software.amazon.awssdk.codegen.poet.PoetExtensions;
+import software.amazon.awssdk.codegen.utils.PaginatorUtils;
+
+public class PaginationDocs {
+
+    private final OperationModel operationModel;
+    private final PoetExtensions poetExtensions;
+
+    public PaginationDocs(IntermediateModel intermediateModel, OperationModel operationModel) {
+        this.operationModel = operationModel;
+        this.poetExtensions = new PoetExtensions(intermediateModel);
+    }
+
+    /**
+     * Constructs additional documentation on the client operation that is appended to the service documentation.
+     */
+    public String getDocsForSyncOperation() {
```

Comment:
Add a TODO here to link to our developer guide. We'll probably want to go into a little more depth in the guide so a link to it when available would be nice.

Reply:
Done